dolphin/processor/config.py

```diff
@@ -508,17 +509,25 @@ def get_index_source_light_model_list(self):
          (for multiple filters)
 
         """
-        if 'lens_light' in self.settings['model']:
-            index_source_light_model_list = []
-            index_num = 0
-            for i in range(self.band_number):
-                single_index_list = []
-                for j in range(len(self.settings['model']['source_light'])):
-                    single_index_list.append(index_num)
-                    index_num += 1
-                index_source_light_model_list.append(single_index_list)
-            return index_source_light_model_list
-
+        if 'source_light' in self.settings['model']:
+            if self.band_number == 1:
+                index_list = [[]]
+                for k, model in enumerate(
+                        self.settings['model']['source_light']):
+                    index_list[0].append(k)
+                return index_list
+            else:
+                if 'source_light_band_index' in self.settings['model']:
+                    index_list_settings =\
+                        self.settings['model']['source_light_band_index']
+                    index_list = [[] for _ in range(self.band_number)]
+                    for i, model in enumerate(index_list_settings):
+                        index_list[model].append(i)
+                    for k in index_list:
+                        assert k != [], "One of the bands have no source light"
+                    return index_list
+                else:
+                    raise ValueError('Missing source_light_band_index')
```

Comment:
Add more description to the error message as `'Missing "source_light_band_indices" in the settings file!'`